packages/core/src/http-handler.ts

```diff
@@ -60,6 +67,23 @@ export class HttpHandler {
     return requestContext;
   }
 
+  protected _bindAuthenticationFn(context: Context): void {
+    context.bind('authenticate').toDynamicValue((): Authenticate => {
+      return async (request: ParsedRequest) => {
+        const meta: AuthenticationMetadata =
+          await context.get(BindingKeys.Context.CONTROLLER_METHOD_META);
+        if (!meta)
+          return;
+        const strategy = context.get(meta.strategy);
```

Comment:
The authentication package has implementation around using passport strategies, There is no need to generalize it any further. Communities/Users can add their own implementation in a separate package and bind it.